File: repls/blastshell-common/src/main/java/blast/shell/repls/RegexBeanSelector.java
package blast.shell.repls;

import java.util.ArrayList;
import java.util.List;
import java.util.regex.Pattern;

/**
 *  Patterns are implicitly "OR'd" -- that is, a bean has to match only one pattern.
 *
 */
public class RegexBeanSelector implements BeanSelector {
    
    List<Pattern> patterns;
    
    @Override
    public boolean filter(String beanName, Class clazz, Object bean) {
        if (patterns != null) {
            for (Pattern pattern : patterns) {
                if (pattern.matcher(beanName).matches()) {
                    return true;
                }
            }
        }
        return false;
    }
    
    public void setPatterns(List<String> strings) {
        this.patterns = new ArrayList<Pattern>();
        for (String pattern : strings) {
            patterns.add(Pattern.compile(pattern));
        }
    }
}

File: repls/blastshell-scala/src/main/scala/blast/shell/repls/RegexBeanSelector.scala
package blast.shell.repls

import java.util.regex.Pattern

/**
 *  Patterns are implicitly "OR'd" -- that is, a bean has to match only one pattern.
 *
 */
class RegexBeanSelector(val regexes: Seq[String]) extends BeanSelector {

  val patterns: Seq[Pattern] = regexes.map {
    pattern: String => Pattern.compile(pattern)
  }

  @Override
  override def filter(beanName: String, clazz: Class[_], bean: Any): Boolean = {
    if (patterns != null) {
      for (pattern <- patterns) {
        if (pattern.matcher(beanName).matches()) {
          true;
        }
      }
    }
    false;
  }

}
